src/DataTypes/DataTypesDecimal.cpp

```diff
@@ -74,19 +74,28 @@ SerializationPtr DataTypeDecimal<T>::doGetDefaultSerialization() const
 
 static DataTypePtr create(const ASTPtr & arguments)
 {
-    if (!arguments || arguments->children.size() != 2)
-        throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
-                        "Decimal data type family must have exactly two arguments: precision and scale");
-
-    const auto * precision = arguments->children[0]->as<ASTLiteral>();
-    const auto * scale = arguments->children[1]->as<ASTLiteral>();
-
-    if (!precision || precision->value.getType() != Field::Types::UInt64 ||
-        !scale || !(scale->value.getType() == Field::Types::Int64 || scale->value.getType() == Field::Types::UInt64))
-        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Decimal data type family must have two numbers as its arguments");
-
-    UInt64 precision_value = precision->value.get<UInt64>();
-    UInt64 scale_value = scale->value.get<UInt64>();
+    UInt64 precision_value = 10;
+    UInt64 scale_value = 0;
+    if (arguments)
+    {
+        if (arguments->children.empty() || arguments->children.size() > 2)
+            throw Exception(
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Decimal data type family must have precision and optional scale arguments");
+
+        const auto * precision = arguments->children[0]->as<ASTLiteral>();
+        if (!precision || precision->value.getType() != Field::Types::UInt64)
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Decimal argument precision is invalid");
+        precision_value = precision->value.get<UInt64>();
+
+        if (arguments->children.size() == 2)
+        {
+            const auto * scale = arguments->children[1]->as<ASTLiteral>();
+            if (!scale || !isInt64OrUInt64FieldType(scale->value.getType()))
```

Comment:
(it was like that before, but what's a negative scale? (Int64)

Reply:
> (it was like that before, but what's a negative scale? (Int64)

Casting to `.get<UInt64>()` would make it positive as previous impl